data/lib/core/position.lua

```diff
@@ -50,14 +28,16 @@ function Position:moveUpstairs()
 	self.z = self.z - 1
 
 	local defaultPosition = self + Position.directionOffset[DIRECTION_SOUTH]
-	if not isWalkable(defaultPosition) then
+	local tileDefaultPosition = Tile(defaultPosition)
```

Comment:
No insted of tileDefaultPosition use:
local toTile = Tile(defaultPosition)

And then you can overwrite toTile with Tile(position) on line 39